index.ts

```diff
@@ -50,19 +54,40 @@ function useLocalStorage<T>(
     if (typeof window === "undefined") return;
 
     const updateLocalStorage = () => {
-      if (storedValue !== undefined) {
-        window.localStorage.setItem(key, serializer(storedValue));
+      // Browser ONLY dispatch storage events to other tabs, NOT current tab.
+      // We need to manually dispatch storage event for current tab
+      if (value !== undefined) {
+        const newValue = serializer(value);
+        const oldValue = rawValueRef.current;
+        rawValueRef.current = newValue;
+        window.localStorage.setItem(key, newValue);
+        window.dispatchEvent(
+          new StorageEvent("storage", {
+            storageArea: window.localStorage,
+            url: window.location.href,
+            key,
+            newValue,
+            oldValue,
```

Comment:
`oldValue` is not used anywhere?

Reply:
@franciscop it is not used in useLocalStorage. But it exists in native StorageEvent. So it is better to have it, in case some other libraries need it.